script/upgrades/MU05/MU05Checks.verify.sol

```diff
@@ -0,0 +1,127 @@
+ */
+interface IBrokerWithCasts {
+  function tradingLimitsConfig(bytes32 id) external view returns (TradingLimits.Config memory);
+}
+
+contract MU05ChecksVerify is MU05ChecksBase {
+  using TradingLimits for TradingLimits.Config;
+  using FixidityLib for FixidityLib.Fraction;
+
+  constructor() public {
+    setUp();
+  }
+
+  function run() public {
+    console.log("\nStarting MU05 checks:");
+    MU05Config.MU05 memory config = MU05Config.get(contracts);
+
+    verifyReserveCollateralAssets();
+    verifyTradingLimits(config);
+  }
+
+  function verifyReserveCollateralAssets() internal {
+    console.log("\n== Verifying Reserve Collateral Assets ==");
+
+    require(Reserve(reserveProxy).isCollateralAsset(nativeUSDC), "❗️❌ Reserve collateral asset not set correctly");
+
+    // mint some native USDC to the main reserve in order to verify spending ratios
+    deal(nativeUSDC, reserveProxy, 100e6);
+    verifyCollateralSpendingRatio(nativeUSDC, 1e24);
+    console.log("🟢 Asset: %s successfully added to collateral asset list", nativeUSDC);
+  }
+
+  function verifyTradingLimits(MU05Config.MU05 memory config) internal view {
+    console.log("\n== Verifying TradingLimits changes in Broker ==");
+    IBrokerWithCasts _broker = IBrokerWithCasts(brokerProxy);
+
+    for (uint256 i = 0; i < config.pools.length; i++) {
+      bytes32 exchangeId = getExchangeId(config.pools[i].asset0, config.pools[i].asset1, config.pools[i].isConstantSum);
+      Config.Pool memory poolConfig = config.pools[i];
+
+      bytes32 asset0LimitId = exchangeId ^ bytes32(uint256(uint160(config.pools[i].asset0)));
+      TradingLimits.Config memory asset0ActualLimit = _broker.tradingLimitsConfig(asset0LimitId);
+
+      bytes32 asset1LimitId = exchangeId ^ bytes32(uint256(uint160(config.pools[i].asset1)));
+      TradingLimits.Config memory asset1ActualLimit = _broker.tradingLimitsConfig(asset1LimitId);
+
+      checkTradingLimt(poolConfig.asset0limits, asset0ActualLimit);
+      checkTradingLimt(poolConfig.asset1limits, asset1ActualLimit);
+    }
+
+    console.log("🟢 Trading limits correctly updated for all exchanges 🔒");
+  }
+
+  function verifyCollateralSpendingRatio(address collateralAsset, uint256 expectedRatio) internal {
+    console.log("\n== Verifying Collateral Spending Ratio for %s ==", collateralAsset);
+    // @notice verifying spending ratios by trying to move the allowed, and more than the allowed amount
+    // the variable holding the ratios is private
+
+    address[] memory otherReserveAddresses = Reserve(reserveProxy).getOtherReserveAddresses();
+    address payable otherReserve = address(uint160(otherReserveAddresses[0]));
+    uint256 reserveBalance = Reserve(reserveProxy).getReserveAddressesCollateralAssetBalance(collateralAsset);
+
+    uint256 spendingLimit = FixidityLib.wrap(expectedRatio).multiply(FixidityLib.newFixed(reserveBalance)).fromFixed();
+    uint256 exceedingAmount = spendingLimit + 1;
+
+    vm.prank(partialReserveMultisig);
```

Comment:
was initially confused on why we needed this after we moved away from the partial reserve. What do you think of renaming to `reserveSpender`?

Reply:
i agree sounds better 👍